bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -176,6 +194,29 @@ private void updateChannel(String channelId) {
         if (isLinked(channelId)) {
             State state = null;
             SonnenJsonDataDTO data = serviceCommunication.getBatteryData();
+            // String direction = GetPowerMeterDirection(channelId);
+
+            // The sonnen API has two sub-channels, e.g. 4_1 and 4_2, one representing consumption and the
+            // other production. E.g. 4_1.kwh_imported represents the total production since the
+            // battery was installed.
+            SonnenJsonPowerMeterDataDTO[] dataPM = null;
+            if (config.powerMeter) {
+                dataPM = serviceCommunication.getPowerMeterDatas();
+            }
+
+            if (dataPM != null) {
+                switch (channelId) {
+                    case KWHIMPORTEDSTATE:
+                        state = new QuantityType<Power>(dataPM[1].getKwhImported(), Units.WATT);
+                        update(state, channelId);
+                        break;
+                    case KWHEXPORTEDSTATE:
+                        state = new QuantityType<Power>(dataPM[1].getKwhExported(), Units.WATT);
```

Comment:
Same here.

Reply:
fixed

Reply:
Rolled back. The unit is Watt hours which i receive from the vendor API. Changed it accordingly.